rest_framework/fields.py

```diff
@@ -1432,28 +1432,44 @@ class ChoiceField(Field):
     html_cutoff = None
     html_cutoff_text = _('More than {count} items...')
 
-    def __init__(self, choices, **kwargs):
+    def __init__(self, choices, underlying_field=None, **kwargs):
+        self.underlying_field = underlying_field
         self.choices = choices
         self.html_cutoff = kwargs.pop('html_cutoff', self.html_cutoff)
         self.html_cutoff_text = kwargs.pop('html_cutoff_text', self.html_cutoff_text)
 
         self.allow_blank = kwargs.pop('allow_blank', False)
-
         super().__init__(**kwargs)
 
+    @staticmethod
+    def _hashable_wrapper(value):
+        if isinstance(value, list):
```

Comment:
Not so sure about this part. `to_representation` can clearly return primitive types which are hashable. It also can return a `dict` (`Serializer`s are still `Field`s, we probably should not restrict using serializers returning a dict). Are there other unhashable types that can be returned from `to_representation`? Should we care? Maybe we should fallback to some dumb list-based `dict` implementation if we cannot hash all the choices?